packages/lowerdash/src/types.ts

```diff
@@ -36,6 +36,12 @@ export type FunctionPropertyNames<T> = {
   [K in keyof T]: T[K] extends Function ? K : never
 }[keyof T]
 
+// The instance type of the inner type of a list
+export type InnerClass<T> = T extends (infer R)[] ? (
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  R extends new (...args: any) => any ? InstanceType<R> : never
+) : never
```

Comment:
Two things:
1. The name `InnerClass` is a bit confusing I think. since this takes in a list of classes and creates a type, maybe this can be called `FromClassList`?
2. I don't think we need to use `any` here, this can be
```suggestion
export type FromClassList<T> = T extends (infer R)[] ? (
  R extends new (...args: never[]) => unknown ? InstanceType<R> : never
) : never```